tensorflow_datasets/core/utils/py_utils.py

```diff
@@ -306,7 +306,13 @@ def incomplete_dir(dirname):
   tf.io.gfile.makedirs(tmp_dir)
   try:
     yield tmp_dir
-    tf.io.gfile.rename(tmp_dir, dirname)
+    renamed = False
+    while not renamed:
```

Comment:
This fix feels too hacky and might creates an infinite loop.
I think the `tf.io.gfile.rename` might fails if `dirname` already exists, which may be the sign of another bug.

Reply:
Yes, this is indeed very hacky.
As, I mentioned previously, this might because the rename function fails to acquire a lock while renaming the directory.
To confirm the same, I added a `sleep` statement before the `rename`  function and it ran without any problem.
So, could you suggest a better solution to this problem?